rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -16,7 +17,21 @@ pid {{ .PID }};
 load_module /etc/nginx/modules/ngx_http_geoip2_module.so;
 {{ end }}
 
+{{/* Range over all locations to see if at least one enables ModSecurity by Annotation 
+{{/* (only if ModSecurity is not enabled globally) */}}
+{{ if not $all.Cfg.EnableModsecurity }}
+{{ range $server := $servers }}
+{{ range $location := $server.Locations }}
+{{ if $location.ModSecurity.Enable }}
+{{ $modSecAnnotationFound = true }}
+{{ end }}
+{{ end }}
+{{ end }}
+{{ end }}
```

Comment:
Please introduce a new function (i.e `shouldLoadModSecurityModule($servers)`) in `template.go` and move this logic to there. That way you can have a unit test for it too.

Then below you can just call that function.